### ISSUE TITLE
Scope in @Directive

### ISSUE THREAD

**Issue by LalitAgrawala004** [vsternbach/angular-ts-decorators#30]
Hi 
I am trying to convert an existing directive using @Directive decorator. This existing directive's controller method has $scope injected. When I am trying to inject $scope in decorator class's constructor I am getting provider error.. Any idea how to access $scope in @Directive class..

**Comment by vsternbach** [comment]
Can you give me a more detailed example?

**Comment by LalitAgrawala004** [comment]
I have something like 

```
export default function testDirective($parse){
	'ngInject';

	return {
		restrict: 'AE',
		
		controller: function($scope, $attrs, TestService, $element){
			'ngInject';
			//logic here
		}
	};
}
```
in the controller function, we have $scope, $attrs $element injected.. I am converting it to @Directive decorator class.. how would i achieve this?

I tried to do it as below -
```
@Directive({
	selector: '[test-directive]'
})
export class testDirective implements IDirective {
	public compile(elem: IAugmentedJQuery) {
	 	elem.css('display', 'none');
	}
	constructor(
		private $scope: any
		private $attrs:IAttributes,
		private $element: IAugmentedJQuery,
	) {
		'ngInject';
	}
}
```
but getting provider error in constructor.. So.. I removed the constructor and tried to implement link and controller functions as it is in IDirective interface.. but those are not getting executed only compile function is getting executed..

**Comment by vsternbach** [comment]
Since angular 2+ doesn't have analogs for compile or link functions you should move all of your code into controller, angular-ts-decorators has support for legacy link and compile as follows, if there's compile method defined on directive's class then it will be used, if there's no compile method, but link method, then it will be used(in both cases class itself isn't instantiated, i.e. injection in the constructor isn't executed), if both aren't defined, then the class itself will be used as directive's controller with the injection in it's constructor.

**Comment by vsternbach** [comment]
fixed in [2eef743](https://github.com/vsternbach/angular-ts-decorators/commit/2eef74329294e9f29ff9c4b85a913baa3c160e4c)
